tests/FakeItEasy.Specs/ArgumentMatchingSpecs.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace FakeItEasy.Specs
+{
+    using FluentAssertions;
+    using Xbehave;
+    using Xunit;
+
+    public static class ArgumentMatchingSpecs
+    {
+        public interface IFoo
+        {
+            int Bar(int a);
+
+            int Bar(int a, string b);
+
+            int Bar(int a, string b, bool c);
+
+            int Bar(int a, string b, bool c, object d);
+        }
+
+        [Scenario]
+        [InlineData(null, 0, 0)]
+        [InlineData(null, 0, 1)]
+        [InlineData(null, 0, 2)]
+        [InlineData(null, 0, 10)]
+        public static void WithAnyArguments(IFoo fake, int result, int argument)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IFoo>());
+
+            "And a fake method with 1 parameter is configured to return a value when called with any arguments"
+                .x(() => A.CallTo(() => fake.Bar(0))
+                            .WithAnyArguments()
+                            .Returns(42));
+
+            "When the method is called with any argument"

```

Comment:
perhaps include `argument` in the message so we know which step fails (not that it will)?